app/controllers/appointments_controller.rb

```diff
@@ -1,31 +1,52 @@
+# Appointments Controller
 class AppointmentsController < ApplicationController
-	load_and_authorize_resource
-	def index
-		@appointments = Appointment.all
-	end
+  load_and_authorize_resource
+  def index
+    @appointments = Appointment.all
+    Appointment.remove_appointments(@appointments)
+  end
 
-	def new
-		@patient = Patient.all
-		@doctor = Doctor.all
-		@appointment = Appointment.new
-	end
+  def new
+    @patients = Patient.all
+    @doctors = Doctor.all
+    @appointment = Appointment.new
+  end
 
-	def create
-		modified_params= appointment_params.merge(doctor_id: params[:doctor_id], patient_id: params[:patient_id])
-		@appointment=Appointment.new(modified_params)
-		respond_to do |format|
-	      if @appointment.save
-	        format.html { redirect_to admin_index_path, notice: 'Appointment was successfully created.' }
-	        format.json { render :show, status: :created, location: @appointment }
-	      else
-	        format.html { render :new }
-	        format.json { render json: @appointment.errors, status: :unprocessable_entity }
-	      end
-		end
-	end
+  def create
+    modified_params = appointment_params.merge(doctor_id: params[:doctor_id], patient_id: params[:patient_id], schedule: params[:schedule ])
+    @appointment = Appointment.new(modified_params)
+    if @appointment.bookable?
+      redirect_to admin_index_path, alert: "Sorry,all 4 appointments for today have been booked for Dr.#{@appointment.doctor.name} on Date: #{@appointment.schedule}"
+    else
+      respond_to do |format|
+        if @appointment.save
+          Appointment.check_role(current_user.role, @appointment)
```

Comment:
instead of changing the status after saving, it would be better to have a `modified_params` that includes the proper value of `status`. Then, create the appointment using  `modified_params`